### PR TITLE
[eiger] Initial working spack package.yaml file.

### PR DESCRIPTION
**Pull request by rasolca**
Added a first working version of compiler.yaml and package.yaml for eiger.
These files are not complete but are a good starting point to use spack on eiger.

### CODE REVIEW

**Comment by lucamar** [comment]
ok to test

File: spack/eiger/packages.yaml [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    variants: ~elpa +mpi +openmp
+  elpa:
+    variants: +openmp +optflags
+#  cray-libsci:
+#    buildable: false
+#    externals:
+#    - spec: cray-libsci@20.06.1
+#      modules:
+#      - cray-libsci/20.06.1
+  fftw:
```
Comment (victorusu) [inline spack/eiger/packages.yaml:127]: There is now a cray-fftw package. Instead of using fftw, here one could use `cray-fftw`

**Review by victorusu — CHANGES_REQUESTED** [review]
@rasolca, awesome!
I just have one request to change from `fftw` to `cray-fftw` in the `packages.yaml` file.

File: spack/eiger/packages.yaml (hunk 1 above)
Comment (rasolca) [reply spack/eiger/packages.yaml]: Done!

**Review by victorusu — APPROVED** [review]
`lgtm`

**Pull request merged**